stdlib/http/src/main/java/org/ballerinalang/net/http/HttpUtil.java

```diff
@@ -1309,10 +1312,88 @@ public static void populateSenderConfigurationOptions(SenderConfiguration sender
         }
     }
 
-    private static boolean isInteger(long val) {
+    public static void populateSenderConfigurations(SenderConfiguration senderConfiguration, Struct
+            clientEndpointConfig) {
+        ProxyServerConfiguration proxyServerConfiguration;
+        Struct secureSocket = clientEndpointConfig.getStructField(HttpConstants.ENDPOINT_CONFIG_SECURE_SOCKET);
+
+        if (secureSocket != null) {
+            HttpUtil.populateSSLConfiguration(senderConfiguration, secureSocket);
+        } else {
+            HttpUtil.setDefaultTrustStore(senderConfiguration);
+        }
+        Struct proxy = clientEndpointConfig.getStructField(HttpConstants.PROXY_STRUCT_REFERENCE);
+        if (proxy != null) {
+            String proxyHost = proxy.getStringField(HttpConstants.PROXY_HOST);
+            int proxyPort = (int) proxy.getIntField(HttpConstants.PROXY_PORT);
+            String proxyUserName = proxy.getStringField(HttpConstants.PROXY_USERNAME);
+            String proxyPassword = proxy.getStringField(HttpConstants.PROXY_PASSWORD);
+            try {
+                proxyServerConfiguration = new ProxyServerConfiguration(proxyHost, proxyPort);
+            } catch (UnknownHostException e) {
+                throw new BallerinaConnectorException("Failed to resolve host" + proxyHost, e);
+            }
+            if (!proxyUserName.isEmpty()) {
+                proxyServerConfiguration.setProxyUsername(proxyUserName);
+            }
+            if (!proxyPassword.isEmpty()) {
+                proxyServerConfiguration.setProxyPassword(proxyPassword);
+            }
+            senderConfiguration.setProxyServerConfiguration(proxyServerConfiguration);
```

Comment:
Shall we move the above code to separate method? It seems duplicated

Reply:
There's a deprecated method (bcoz of that the duplication) and that will be removed by @daneshk  and will use this method instead. That change will be done on top of this, hence no point removing this to a separate method